python/lsst/daf/butler/core/storageClass.py

```diff
@@ -842,6 +856,55 @@ def getStorageClass(self, storageClassName: str) -> StorageClass:
         """
         return self._storageClasses[storageClassName]
 
+    def findStorageClass(self, pytype: Type, compare_types: bool = False) -> StorageClass:
+        """Find the storage class associated with this python type.
+
+        Parameters
+        ----------
+        pytype : `type`
+            The Python type to be matched.
+        compare_types : `bool`, optional
+            If `False`, the type will be checked against name of the python
+            type. This comparison is always done first. If `True` and the
+            string comparison failed, each candidate storage class will be
+            forced to have its type imported. This can be significantly slower.
+
+        Returns
+        -------
+        storageClass : `StorageClass`
+            The matching storage class.
+
+        Raises
+        ------
+        KeyError
+            Raised if no match could be found.
+
+        Notes
+        -----
+        It is possible for a python type to be associated with multiple
+        storage classes. This method will currently return the first that
+        matches.
+        """
+        result = self._find_storage_class(pytype, False)
+        if result:
+            return result
+
+        if compare_types:
+            # The fast comparison failed and we were asked to try the
+            # variant that might involve code imports.
+            result = self._find_storage_class(pytype, True)
+            if result:
+                return result
+
+        raise KeyError(f"Unable to find a StorageClass associated with type {get_full_type_name(pytype)}")
+
+    def _find_storage_class(self, pytype: Type, compare_types: bool) -> Optional[StorageClass]:
```

Comment:
Does this need a docstring?

Reply:
It's really meant to be a super secret helper method. I toyed with having it be inside the other method but seemed better to be outside. I don't think we are required to fully document these kinds of methods but I will do so.